File: backend/src/modules/subscriber/subscriber.controller.ts
import {
  Controller,
  Get,
  Post,
  Body,
  Patch,
  Param,
  Delete,
  Query,
  ParseIntPipe,
} from '@nestjs/common';
import { ApiTags, ApiOperation, ApiResponse } from '@nestjs/swagger';
import { SubscriberService } from './subscriber.service';
import { CreateSubscriberDto } from './dto/create-subscriber.dto';
import { UpdateSubscriberDto } from './dto/update-subscriber.dto';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';
import { Subscriber } from './entities/subscriber.entity';

@ApiTags('用户管理')
@Controller('api/subscribers')
export class SubscriberController {
  constructor(private readonly subscriberService: SubscriberService) {}

  @Post()
  @ApiOperation({ summary: '创建用户' })
  @ApiResponse({ status: 201, description: '创建成功' })
  async create(@Body() createDto: CreateSubscriberDto): Promise<ApiResponseDto<Subscriber>> {
    const subscriber = await this.subscriberService.create(createDto);
    return ApiResponseDto.success(subscriber, '创建成功');
  }

  @Get()
  @ApiOperation({ summary: '获取用户列表' })
  @ApiResponse({ status: 200, description: '获取成功' })
  async findAll(@Query() paginationDto: PaginationDto): Promise<ApiResponseDto> {
    const result = await this.subscriberService.findAll(paginationDto);
    return ApiResponseDto.success(result);
  }

  @Get('stats')
  @ApiOperation({ summary: '获取用户统计信息' })
  @ApiResponse({ status: 200, description: '获取成功' })
  async getStats(): Promise<ApiResponseDto> {
    const stats = await this.subscriberService.getStats();
    return ApiResponseDto.success(stats);
  }

  @Get(':id')
  @ApiOperation({ summary: '获取用户详情' })
  @ApiResponse({ status: 200, description: '获取成功' })
  async findOne(@Param('id', ParseIntPipe) id: number): Promise<ApiResponseDto<Subscriber>> {
    const subscriber = await this.subscriberService.findOne(id);
    return ApiResponseDto.success(subscriber);
  }

  @Patch(':id')
  @ApiOperation({ summary: '更新用户' })
  @ApiResponse({ status: 200, description: '更新成功' })
  async update(
    @Param('id', ParseIntPipe) id: number,
    @Body() updateDto: UpdateSubscriberDto,
  ): Promise<ApiResponseDto<Subscriber>> {
    const subscriber = await this.subscriberService.update(id, updateDto);
    return ApiResponseDto.success(subscriber, '更新成功');
  }

  @Delete(':id')
  @ApiOperation({ summary: '删除用户' })
  @ApiResponse({ status: 200, description: '删除成功' })
  async remove(@Param('id', ParseIntPipe) id: number): Promise<ApiResponseDto> {
    await this.subscriberService.remove(id);
    return ApiResponseDto.success(null, '删除成功');
  }

  @Post('batch-delete')
  @ApiOperation({ summary: '批量删除用户' })
  @ApiResponse({ status: 200, description: '删除成功' })
  async batchRemove(@Body('ids') ids: number[]): Promise<ApiResponseDto> {
    await this.subscriberService.batchRemove(ids);
    return ApiResponseDto.success(null, '删除成功');
  }
}

File: kam-admin-module-user/src/main/java/io/kamailio/admin/modules/subscriber/controller/SubscriberController.java
package io.kamailio.admin.modules.subscriber.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.subscriber.dto.CreateSubscriberDto;
import io.kamailio.admin.modules.subscriber.dto.UpdateSubscriberDto;
import io.kamailio.admin.modules.subscriber.entity.Subscriber;
import io.kamailio.admin.modules.subscriber.service.SubscriberService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@Tag(name = "用户管理")
@RestController
@RequestMapping("/api/subscribers")
public class SubscriberController {

    private final SubscriberService service;

    public SubscriberController(SubscriberService service) {
        this.service = service;
    }

    @PostMapping
    @Operation(summary = "创建用户")
    public ResponseEntity<ApiResponse<Subscriber>> create(@Valid @RequestBody CreateSubscriberDto dto) {
        var entity = service.create(dto);
        return ResponseEntity.ok(ApiResponse.success(entity, "创建成功"));
    }

    @GetMapping
    @Operation(summary = "获取用户列表")
    public ResponseEntity<ApiResponse<PaginatedResult<Subscriber>>> findAll(PaginationDto dto) {
        var result = service.findAll(dto);
        return ResponseEntity.ok(ApiResponse.success(result));
    }

    @GetMapping("/stats")
    @Operation(summary = "获取用户统计信息")
    public ResponseEntity<ApiResponse<Map<String, Object>>> getStats() {
        return ResponseEntity.ok(ApiResponse.success(service.getStats()));
    }

    @GetMapping("/{id}")
    @Operation(summary = "获取用户详情")
    public ResponseEntity<ApiResponse<Subscriber>> findOne(@PathVariable Integer id) {
        return ResponseEntity.ok(ApiResponse.success(service.findOne(id)));
    }

    @PatchMapping("/{id}")
    @Operation(summary = "更新用户")
    public ResponseEntity<ApiResponse<Subscriber>> update(@PathVariable Integer id, @Valid @RequestBody UpdateSubscriberDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.update(id, dto), "更新成功"));
    }

    @DeleteMapping("/{id}")
    @Operation(summary = "删除用户")
    public ResponseEntity<ApiResponse<Void>> remove(@PathVariable Integer id) {
        service.remove(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    @PostMapping("/batch-delete")
    @Operation(summary = "批量删除用户")
    public ResponseEntity<ApiResponse<Void>> batchRemove(@RequestBody BatchDeleteRequest body) {
        service.batchRemove(body.getIds());
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }

    public static class BatchDeleteRequest {
        private List<Integer> ids;
        public List<Integer> getIds() { return ids; }
        public void setIds(List<Integer> ids) { this.ids = ids; }
    }
}
